{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5d4e725b-9724-43e3-b947-cd42b1091e5b",
   "metadata": {},
   "source": [
    "Sequoia package for kinship reconstruction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e4ce11e8-6b22-48e3-9882-ef3cf84e8c30",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Make sure R can find packages installed through Conda\n",
    ".libPaths('/hb/home/jbos/.conda/envs/vcfR')\n",
    ".libPaths(\"/hb/home/jbos/.conda/envs/vcfR/lib/R/library\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d4a4aa5a-0d42-4acf-968a-4d1c9115d29f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "   *****       ***   vcfR   ***       *****\n",
      "   This is vcfR 1.15.0 \n",
      "     browseVignettes('vcfR') # Documentation\n",
      "     citation('vcfR') # Citation\n",
      "   *****       *****      *****       *****\n",
      "\n",
      "\n",
      "Linking to GEOS 3.12.2, GDAL 3.9.2, PROJ 9.5.0; sf_use_s2() is TRUE\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(sequoia)\n",
    "library(vcfR)\n",
    "library(sf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b8f47ea5-2570-4b6f-9512-26cf1042ab09",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Function to make input for Sequoia, despite the fact that our samples don't have sex or birth year\n",
    "makeLifeHistory<-function(vcf){\n",
    "    lh<-as.data.frame(colnames(vcf@gt)[2:length(colnames(vcf@gt))])\n",
    "    lh$Sex<-4\n",
    "    lh$BirthYear<--1\n",
    "    colnames(lh)<-c('ID','Sex','BirthYear')\n",
    "    return(lh)\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "467fac6f-11d1-4f27-bbae-83c2f367e01f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Calculate great circle distances from lat/lons\n",
    "wide_dists <- function(sites){\n",
    "    x<-st_as_sf(sites,coords=c('LONGITUDE','LATITUDE'))\n",
    "    x<-st_set_crs(x,4326)\n",
    "    d <- st_distance(x)\n",
    "    colnames(d)<-sites$Name\n",
    "    rownames(d)<-sites$Name\n",
    "    d<-as.data.frame(d)\n",
    "    return(d)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "6cad6de0-2d5c-44da-a185-59cf1c197a41",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Check distance between potential relatives\n",
    "checkDist<-function(mayberel,dists,PairNumber){\n",
    "    pn<-PairNumber\n",
    "    p<-unlist(strsplit(mayberel$MaybePar$ID1[pn],\"_\"))[1]\n",
    "    o<-unlist(strsplit(mayberel$MaybePar$ID2[pn],\"_\"))[1]\n",
    "    return(dists[p,o])\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c8b0496e-c11c-4d86-94b3-8e6d623cdb00",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in GenoConvert(InFile = \"/hb/scratch/jbos/spp1/vcf_thinned500bp.recode.vcf\", : rows have unequal length\n",
     "output_type": "error",
     "traceback": [
      "Error in GenoConvert(InFile = \"/hb/scratch/jbos/spp1/vcf_thinned500bp.recode.vcf\", : rows have unequal length\nTraceback:\n",
      "1. stop(\"rows have unequal length\")",
      "2. .handleSimpleError(function (cnd) \n . {\n .     watcher$capture_plot_and_output()\n .     cnd <- sanitize_call(cnd)\n .     watcher$push(cnd)\n .     switch(on_error, continue = invokeRestart(\"eval_continue\"), \n .         stop = invokeRestart(\"eval_stop\"), error = NULL)\n . }, \"rows have unequal length\", base::quote(GenoConvert(InFile = \"/hb/scratch/jbos/spp1/vcf_thinned500bp.recode.vcf\", \n .     OutFile = \"snps1.txt\", OutFormat = \"seq\")))"
     ]
    }
   ],
   "source": [
    "#Create SNPs.txt file for each taxon\n",
    "sequoia1<-GenoConvert(InFile=\"/hb/scratch/jbos/spp1/vcf_thinned500bp.recode.vcf\",OutFile='snps1.txt',OutFormat='seq')\n",
    "sequoia2<-GenoConvert(InFile=\"/hb/scratch/jbos/spp2/vcf_thinned500bp.recode.vcf\",OutFile='snps2.txt',OutFormat='seq')\n",
    "sequoia3<-GenoConvert(InFile=\"/hb/scratch/jbos/spp3/vcf_thinned500bp.recode.vcf\",OutFile='snps3.txt',OutFormat='seq')\n",
    "sequoia4<-GenoConvert(InFile=\"/hb/scratch/jbos/spp4/vcf_thinned500bp.recode.vcf\",OutFile='snps4.txt',OutFormat='seq')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8177db7e-1e3e-4099-bb06-29cef1d436b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scanning file to determine attributes.\n",
      "File attributes:\n",
      "  meta lines: 1605\n",
      "  header_line: 1606\n",
      "  variant count: 606\n",
      "  column count: 133\n",
      "Meta line 1605 read in.\n",
      "All meta lines processed.\n",
      "gt matrix initialized.\n",
      "Character matrix gt created.\n",
      "  Character matrix gt rows: 606\n",
      "  Character matrix gt cols: 133\n",
      "  skip: 0\n",
      "  nrows: 606\n",
      "  row_num: 0\n",
      "Processed variant: 606\n",
      "All variants processed\n",
      "Scanning file to determine attributes.\n",
      "File attributes:\n",
      "  meta lines: 1605\n",
      "  header_line: 1606\n",
      "  variant count: 640\n",
      "  column count: 82\n",
      "Meta line 1605 read in.\n",
      "All meta lines processed.\n",
      "gt matrix initialized.\n",
      "Character matrix gt created.\n",
      "  Character matrix gt rows: 640\n",
      "  Character matrix gt cols: 82\n",
      "  skip: 0\n",
      "  nrows: 640\n",
      "  row_num: 0\n",
      "Processed variant: 640\n",
      "All variants processed\n",
      "Scanning file to determine attributes.\n",
      "File attributes:\n",
      "  meta lines: 1605\n",
      "  header_line: 1606\n",
      "  variant count: 798\n",
      "  column count: 69\n",
      "Meta line 1605 read in.\n",
      "All meta lines processed.\n",
      "gt matrix initialized.\n",
      "Character matrix gt created.\n",
      "  Character matrix gt rows: 798\n",
      "  Character matrix gt cols: 69\n",
      "  skip: 0\n",
      "  nrows: 798\n",
      "  row_num: 0\n",
      "Processed variant: 798\n",
      "All variants processed\n",
      "Scanning file to determine attributes.\n",
      "File attributes:\n",
      "  meta lines: 1605\n",
      "  header_line: 1606\n",
      "  variant count: 435\n",
      "  column count: 78\n",
      "Meta line 1605 read in.\n",
      "All meta lines processed.\n",
      "gt matrix initialized.\n",
      "Character matrix gt created.\n",
      "  Character matrix gt rows: 435\n",
      "  Character matrix gt cols: 78\n",
      "  skip: 0\n",
      "  nrows: 435\n",
      "  row_num: 0\n",
      "Processed variant: 435\n",
      "All variants processed\n"
     ]
    }
   ],
   "source": [
    "#Read in SNP data for each taxon\n",
    "vcf1<-read.vcfR(\"/hb/scratch/jbos/spp1/vcf_thinned500bp.recode.vcf\")\n",
    "vcf2<-read.vcfR(\"/hb/scratch/jbos/spp2/vcf_thinned500bp.recode.vcf\")\n",
    "vcf3<-read.vcfR(\"/hb/scratch/jbos/spp3/vcf_thinned500bp.recode.vcf\")\n",
    "vcf4<-read.vcfR(\"/hb/scratch/jbos/spp4/vcf_thinned500bp.recode.vcf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "6b178d7d-5629-4121-9ff5-1ac99f80c50d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Make mostly empty life history inputs for Sequoia functions\n",
    "lh1<-makeLifeHistory(vcf1)\n",
    "lh2<-makeLifeHistory(vcf2)\n",
    "lh3<-makeLifeHistory(vcf3)\n",
    "lh4<-makeLifeHistory(vcf4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "cc33b225-4600-4533-a641-05eb1ef462c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Make genome inputs\n",
    "sequoia1<-as.matrix(read.table(\"snps1.txt\", row.names=1, header=FALSE))\n",
    "sequoia2<-as.matrix(read.table(\"snps2.txt\", row.names=1, header=FALSE))\n",
    "sequoia3<-as.matrix(read.table(\"snps3.txt\", row.names=1, header=FALSE))\n",
    "sequoia4<-as.matrix(read.table(\"snps4.txt\", row.names=1, header=FALSE))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "fba4f47b-ada5-4f51-a4f9-440adc53069a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Searching for non-assigned parent-offspring pairs ... (Module = par)\n",
      "\n",
      "Warning message:\n",
      "“ There are 2 monomorphic (fixed) SNPs, these will be excluded \n",
      "”\n",
      "After exclusion, There are  124  individuals and  549  SNPs.\n",
      "\n",
      "\n",
      "Ageprior: Flat 0/1, overlapping generations, MaxAgeParent = 99,99\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Counting opposing homozygous loci between all individuals ...\n",
      "Checking for non-assigned Parent-Offspring pairs ...\n",
      "Checking for Parent-Parent-Offspring trios ...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found 2 likely parent-offspring pairs, and 0 other non-assigned pairs of possible relatives\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Find potential relatives with no age prior\n",
    "MaybeRel1 <- GetMaybeRel(GenoM = sequoia1, Err = 0.005, LifeHistData = lh1, quiet=FALSE,Herm='B')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "bbd441d5-fbc2-41d1-b1eb-f13fa4fbf4ae",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Searching for non-assigned parent-offspring pairs ... (Module = par)\n",
      "\n",
      "Warning message:\n",
      "“ There are 1 monomorphic (fixed) SNPs, these will be excluded \n",
      "”\n",
      "After exclusion, There are  73  individuals and  602  SNPs.\n",
      "\n",
      "\n",
      "Ageprior: Flat 0/1, overlapping generations, MaxAgeParent = 99,99\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Counting opposing homozygous loci between all individuals ...\n",
      "Checking for non-assigned Parent-Offspring pairs ...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found 1 likely parent-offspring pairs, and 0 other non-assigned pairs of possible relatives\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Find potential relatives with no age prior\n",
    "MaybeRel2 <- GetMaybeRel(GenoM = sequoia2, Err = 0.005, LifeHistData = lh2, quiet=FALSE,Herm='B')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "1d39bb44-a2a7-4e21-bfa4-cee5964f466b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Searching for non-assigned parent-offspring pairs ... (Module = par)\n",
      "\n",
      "Genotype matrix looks OK! There are  60  individuals and  750  SNPs.\n",
      "\n",
      "\n",
      "Ageprior: Flat 0/1, overlapping generations, MaxAgeParent = 99,99\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Counting opposing homozygous loci between all individuals ...\n",
      "Checking for non-assigned Parent-Offspring pairs ...\n",
      "Checking for Parent-Parent-Offspring trios ...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found 3 likely parent-offspring pairs, and 0 other non-assigned pairs of possible relatives\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Find potential relatives with no age prior\n",
    "MaybeRel3 <- GetMaybeRel(GenoM = sequoia3, Err = 0.005, LifeHistData = lh3, quiet=FALSE,Herm='B')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "5be13bd7-4e93-4534-8324-90465e5fa0e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Searching for non-assigned parent-offspring pairs ... (Module = par)\n",
      "\n",
      "Genotype matrix looks OK! There are  69  individuals and  404  SNPs.\n",
      "\n",
      "\n",
      "Ageprior: Flat 0/1, overlapping generations, MaxAgeParent = 99,99\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Counting opposing homozygous loci between all individuals ...\n",
      "Checking for non-assigned Parent-Offspring pairs ...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found 1 likely parent-offspring pairs, and 0 other non-assigned pairs of possible relatives\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Find potential relatives with no age prior\n",
    "MaybeRel4 <- GetMaybeRel(GenoM = sequoia4, Err = 0.005, LifeHistData = lh4, quiet=FALSE,Herm='B')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "0514f626-fa77-4439-bed5-b193433306dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$MaybePar</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID1</th><th scope=col>ID2</th><th scope=col>TopRel</th><th scope=col>LLR</th><th scope=col>OH</th><th scope=col>BirthYear1</th><th scope=col>BirthYear2</th><th scope=col>AgeDif</th><th scope=col>Sex1</th><th scope=col>Sex2</th><th scope=col>SNPdBoth</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>LEY11_T15</td><td>LEY04_T15</td><td>PO</td><td>5.30</td><td>0</td><td>NA</td><td>NA</td><td>NA</td><td>4</td><td>4</td><td>544</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>LEY07_T02</td><td>CEB35_T08</td><td>PO</td><td>1.78</td><td>0</td><td>NA</td><td>NA</td><td>NA</td><td>4</td><td>4</td><td>367</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$MaybeTrio</dt>\n",
       "\t\t<dd>NULL</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$MaybePar] A data.frame: 2 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & ID1 & ID2 & TopRel & LLR & OH & BirthYear1 & BirthYear2 & AgeDif & Sex1 & Sex2 & SNPdBoth\\\\\n",
       "  & <chr> & <chr> & <chr> & <dbl> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & LEY11\\_T15 & LEY04\\_T15 & PO & 5.30 & 0 & NA & NA & NA & 4 & 4 & 544\\\\\n",
       "\t2 & LEY07\\_T02 & CEB35\\_T08 & PO & 1.78 & 0 & NA & NA & NA & 4 & 4 & 367\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$MaybeTrio] NULL\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$MaybePar\n",
       ":   \n",
       "A data.frame: 2 × 11\n",
       "\n",
       "| <!--/--> | ID1 &lt;chr&gt; | ID2 &lt;chr&gt; | TopRel &lt;chr&gt; | LLR &lt;dbl&gt; | OH &lt;int&gt; | BirthYear1 &lt;dbl&gt; | BirthYear2 &lt;dbl&gt; | AgeDif &lt;dbl&gt; | Sex1 &lt;dbl&gt; | Sex2 &lt;dbl&gt; | SNPdBoth &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | LEY11_T15 | LEY04_T15 | PO | 5.30 | 0 | NA | NA | NA | 4 | 4 | 544 |\n",
       "| 2 | LEY07_T02 | CEB35_T08 | PO | 1.78 | 0 | NA | NA | NA | 4 | 4 | 367 |\n",
       "\n",
       "\n",
       "$MaybeTrio\n",
       ":   NULL\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$MaybePar\n",
       "        ID1       ID2 TopRel  LLR OH BirthYear1 BirthYear2 AgeDif Sex1 Sex2\n",
       "1 LEY11_T15 LEY04_T15     PO 5.30  0         NA         NA     NA    4    4\n",
       "2 LEY07_T02 CEB35_T08     PO 1.78  0         NA         NA     NA    4    4\n",
       "  SNPdBoth\n",
       "1      544\n",
       "2      367\n",
       "\n",
       "$MaybeTrio\n",
       "NULL\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "MaybeRel1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "77169ac9-d5ba-412e-b639-f2bdeb1cfe22",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$MaybePar</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID1</th><th scope=col>ID2</th><th scope=col>TopRel</th><th scope=col>LLR</th><th scope=col>OH</th><th scope=col>BirthYear1</th><th scope=col>BirthYear2</th><th scope=col>AgeDif</th><th scope=col>Sex1</th><th scope=col>Sex2</th><th scope=col>SNPdBoth</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>LEY14_T09</td><td>CEB07_T14</td><td>PO</td><td>8.62</td><td>0</td><td>NA</td><td>NA</td><td>NA</td><td>4</td><td>4</td><td>591</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$MaybeTrio</dt>\n",
       "\t\t<dd>NULL</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$MaybePar] A data.frame: 1 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & ID1 & ID2 & TopRel & LLR & OH & BirthYear1 & BirthYear2 & AgeDif & Sex1 & Sex2 & SNPdBoth\\\\\n",
       "  & <chr> & <chr> & <chr> & <dbl> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & LEY14\\_T09 & CEB07\\_T14 & PO & 8.62 & 0 & NA & NA & NA & 4 & 4 & 591\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$MaybeTrio] NULL\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$MaybePar\n",
       ":   \n",
       "A data.frame: 1 × 11\n",
       "\n",
       "| <!--/--> | ID1 &lt;chr&gt; | ID2 &lt;chr&gt; | TopRel &lt;chr&gt; | LLR &lt;dbl&gt; | OH &lt;int&gt; | BirthYear1 &lt;dbl&gt; | BirthYear2 &lt;dbl&gt; | AgeDif &lt;dbl&gt; | Sex1 &lt;dbl&gt; | Sex2 &lt;dbl&gt; | SNPdBoth &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | LEY14_T09 | CEB07_T14 | PO | 8.62 | 0 | NA | NA | NA | 4 | 4 | 591 |\n",
       "\n",
       "\n",
       "$MaybeTrio\n",
       ":   NULL\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$MaybePar\n",
       "        ID1       ID2 TopRel  LLR OH BirthYear1 BirthYear2 AgeDif Sex1 Sex2\n",
       "1 LEY14_T09 CEB07_T14     PO 8.62  0         NA         NA     NA    4    4\n",
       "  SNPdBoth\n",
       "1      591\n",
       "\n",
       "$MaybeTrio\n",
       "NULL\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "MaybeRel2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "100278b4-c8f0-47ec-8a2c-dc08cd33b6ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$MaybePar</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID1</th><th scope=col>ID2</th><th scope=col>TopRel</th><th scope=col>LLR</th><th scope=col>OH</th><th scope=col>BirthYear1</th><th scope=col>BirthYear2</th><th scope=col>AgeDif</th><th scope=col>Sex1</th><th scope=col>Sex2</th><th scope=col>SNPdBoth</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>CEB01_T12</td><td>CEB23_T16</td><td>PO</td><td>6.52</td><td>0</td><td>NA</td><td>NA</td><td>NA</td><td>4</td><td>4</td><td>681</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>CEB01_T11</td><td>CEB01_T12</td><td>PO</td><td>6.04</td><td>0</td><td>NA</td><td>NA</td><td>NA</td><td>4</td><td>4</td><td>719</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>CEB01_T11</td><td>CEB23_T16</td><td>PO</td><td>5.39</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>4</td><td>4</td><td>676</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$MaybeTrio</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>id</th><th scope=col>parent1</th><th scope=col>parent2</th><th scope=col>LLRparent1</th><th scope=col>LLRparent2</th><th scope=col>LLRpair</th><th scope=col>OHparent1</th><th scope=col>OHparent2</th><th scope=col>MEpair</th><th scope=col>SNPd.id.parent1</th><th scope=col>SNPd.id.parent2</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>CEB23_T16</td><td>CEB01_T11</td><td>CEB01_T12</td><td>-0.3</td><td>-3.89</td><td>0.28</td><td>1</td><td>0</td><td>5</td><td>676</td><td>681</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$MaybePar] A data.frame: 3 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & ID1 & ID2 & TopRel & LLR & OH & BirthYear1 & BirthYear2 & AgeDif & Sex1 & Sex2 & SNPdBoth\\\\\n",
       "  & <chr> & <chr> & <chr> & <dbl> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & CEB01\\_T12 & CEB23\\_T16 & PO & 6.52 & 0 & NA & NA & NA & 4 & 4 & 681\\\\\n",
       "\t2 & CEB01\\_T11 & CEB01\\_T12 & PO & 6.04 & 0 & NA & NA & NA & 4 & 4 & 719\\\\\n",
       "\t3 & CEB01\\_T11 & CEB23\\_T16 & PO & 5.39 & 1 & NA & NA & NA & 4 & 4 & 676\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$MaybeTrio] A data.frame: 1 × 11\n",
       "\\begin{tabular}{lllllllllll}\n",
       " id & parent1 & parent2 & LLRparent1 & LLRparent2 & LLRpair & OHparent1 & OHparent2 & MEpair & SNPd.id.parent1 & SNPd.id.parent2\\\\\n",
       " <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t CEB23\\_T16 & CEB01\\_T11 & CEB01\\_T12 & -0.3 & -3.89 & 0.28 & 1 & 0 & 5 & 676 & 681\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$MaybePar\n",
       ":   \n",
       "A data.frame: 3 × 11\n",
       "\n",
       "| <!--/--> | ID1 &lt;chr&gt; | ID2 &lt;chr&gt; | TopRel &lt;chr&gt; | LLR &lt;dbl&gt; | OH &lt;int&gt; | BirthYear1 &lt;dbl&gt; | BirthYear2 &lt;dbl&gt; | AgeDif &lt;dbl&gt; | Sex1 &lt;dbl&gt; | Sex2 &lt;dbl&gt; | SNPdBoth &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | CEB01_T12 | CEB23_T16 | PO | 6.52 | 0 | NA | NA | NA | 4 | 4 | 681 |\n",
       "| 2 | CEB01_T11 | CEB01_T12 | PO | 6.04 | 0 | NA | NA | NA | 4 | 4 | 719 |\n",
       "| 3 | CEB01_T11 | CEB23_T16 | PO | 5.39 | 1 | NA | NA | NA | 4 | 4 | 676 |\n",
       "\n",
       "\n",
       "$MaybeTrio\n",
       ":   \n",
       "A data.frame: 1 × 11\n",
       "\n",
       "| id &lt;chr&gt; | parent1 &lt;chr&gt; | parent2 &lt;chr&gt; | LLRparent1 &lt;dbl&gt; | LLRparent2 &lt;dbl&gt; | LLRpair &lt;dbl&gt; | OHparent1 &lt;int&gt; | OHparent2 &lt;int&gt; | MEpair &lt;int&gt; | SNPd.id.parent1 &lt;int&gt; | SNPd.id.parent2 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| CEB23_T16 | CEB01_T11 | CEB01_T12 | -0.3 | -3.89 | 0.28 | 1 | 0 | 5 | 676 | 681 |\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$MaybePar\n",
       "        ID1       ID2 TopRel  LLR OH BirthYear1 BirthYear2 AgeDif Sex1 Sex2\n",
       "1 CEB01_T12 CEB23_T16     PO 6.52  0         NA         NA     NA    4    4\n",
       "2 CEB01_T11 CEB01_T12     PO 6.04  0         NA         NA     NA    4    4\n",
       "3 CEB01_T11 CEB23_T16     PO 5.39  1         NA         NA     NA    4    4\n",
       "  SNPdBoth\n",
       "1      681\n",
       "2      719\n",
       "3      676\n",
       "\n",
       "$MaybeTrio\n",
       "         id   parent1   parent2 LLRparent1 LLRparent2 LLRpair OHparent1\n",
       "1 CEB23_T16 CEB01_T11 CEB01_T12       -0.3      -3.89    0.28         1\n",
       "  OHparent2 MEpair SNPd.id.parent1 SNPd.id.parent2\n",
       "1         0      5             676             681\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "MaybeRel3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "b66bf019-d77f-4fb1-bdb0-e580a01af066",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$MaybePar</dt>\n",
       "\t\t<dd><table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID1</th><th scope=col>ID2</th><th scope=col>TopRel</th><th scope=col>LLR</th><th scope=col>OH</th><th scope=col>BirthYear1</th><th scope=col>BirthYear2</th><th scope=col>AgeDif</th><th scope=col>Sex1</th><th scope=col>Sex2</th><th scope=col>SNPdBoth</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>CEB26_T22</td><td>CEB05_T12</td><td>PO</td><td>2.97</td><td>0</td><td>NA</td><td>NA</td><td>NA</td><td>4</td><td>4</td><td>396</td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$MaybeTrio</dt>\n",
       "\t\t<dd>NULL</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$MaybePar] A data.frame: 1 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & ID1 & ID2 & TopRel & LLR & OH & BirthYear1 & BirthYear2 & AgeDif & Sex1 & Sex2 & SNPdBoth\\\\\n",
       "  & <chr> & <chr> & <chr> & <dbl> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & CEB26\\_T22 & CEB05\\_T12 & PO & 2.97 & 0 & NA & NA & NA & 4 & 4 & 396\\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$MaybeTrio] NULL\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$MaybePar\n",
       ":   \n",
       "A data.frame: 1 × 11\n",
       "\n",
       "| <!--/--> | ID1 &lt;chr&gt; | ID2 &lt;chr&gt; | TopRel &lt;chr&gt; | LLR &lt;dbl&gt; | OH &lt;int&gt; | BirthYear1 &lt;dbl&gt; | BirthYear2 &lt;dbl&gt; | AgeDif &lt;dbl&gt; | Sex1 &lt;dbl&gt; | Sex2 &lt;dbl&gt; | SNPdBoth &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | CEB26_T22 | CEB05_T12 | PO | 2.97 | 0 | NA | NA | NA | 4 | 4 | 396 |\n",
       "\n",
       "\n",
       "$MaybeTrio\n",
       ":   NULL\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$MaybePar\n",
       "        ID1       ID2 TopRel  LLR OH BirthYear1 BirthYear2 AgeDif Sex1 Sex2\n",
       "1 CEB26_T22 CEB05_T12     PO 2.97  0         NA         NA     NA    4    4\n",
       "  SNPdBoth\n",
       "1      396\n",
       "\n",
       "$MaybeTrio\n",
       "NULL\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "MaybeRel4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "76543e7f-57ce-4dfa-869c-83805c41ff67",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Import site locations with lat/lons\n",
    "sites <- read.csv(\"all_Atenuis_sites_FIXED.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "be51b2fc-8446-4156-b922-4b1fdb0a9ba0",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create distance matrix from sites\n",
    "dists<-wide_dists(sites)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "091efe88-126c-43d6-a35c-2a0723b0fb48",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "58.33163 [m]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "44.55889 [m]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Check distance between relatives and convert to km, taxa 1\n",
    "checkDist(MaybeRel1,dists,1)/1000\n",
    "checkDist(MaybeRel1,dists,2)/1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "6f556b47-3eac-42fc-9ccc-0dda97da573a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "142.4513 [m]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Check distance between relatives and convert to km, taxa 2\n",
    "checkDist(MaybeRel2,dists,1)/1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "f4c8e9c1-6b9c-4298-b41b-3195a6dd5273",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "180.5587 [m]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "0 [m]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "180.5587 [m]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Check distance between relatives and convert to km, taxa 3\n",
    "checkDist(MaybeRel3,dists,1)/1000\n",
    "checkDist(MaybeRel3,dists,2)/1000\n",
    "checkDist(MaybeRel3,dists,3)/1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "a7c83d7b-a057-44fc-b09c-bf20625788e5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "174.1064 [m]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Check distance between relatives and convert to km, taxa 4\n",
    "checkDist(MaybeRel4,dists,1)/1000"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
